src/services/codefixes/fixExpectedCommaError.ts

```diff
@@ -0,0 +1,47 @@
+/* @internal */
+namespace ts.codefix {
+    const fixId = "fixExpectedCommaError";
+    const expectedErrorCode = Diagnostics._0_expected.code;
+    const errorCodes = [expectedErrorCode];
+
+    registerCodeFix({
+        errorCodes,
+        getCodeActions(context) {
+            const { sourceFile } = context;
+            const info = getInfo(sourceFile, context.span.start, context.errorCode);
+            if (!info) { return undefined; }
+
+            const changes = textChanges.ChangeTracker.with(context, t => doChange(t, sourceFile, info));
+
+            return [createCodeFixAction(
+                fixId,
+                changes,
+                [Diagnostics.Change_0_to_1, ';', ','],
+                fixId,
+                [Diagnostics.Change_0_to_1, ';', ',']
+            )];
+        },
+        fixIds: [fixId],
+        getAllCodeActions: context => codeFixAll(context, errorCodes, (changes, diag) => {
+            const info = getInfo(diag.file, diag.start, diag.code);
+            if (info) doChange(changes, context.sourceFile, info);
+        }),
+    });
+
+    interface Info { readonly node: Node; }
+
+    function getInfo(sourceFile: SourceFile, pos: number, _: number): Info | undefined {
+        const node = getTokenAtPosition(sourceFile, pos);
+        const isCorrectContext = isObjectLiteralExpression(node.parent) ||
+                                 isArrayLiteralExpression(node.parent);
+
+        return (node.kind === SyntaxKind.SemicolonToken &&
+                node.parent &&

```

Comment:
Can you move this check up to where you declare `isCorrectContext` so that it checks that `node.parent` is defined before it checks the type of node?